dove/src/transaction.rs

```diff
@@ -0,0 +1,875 @@
+                .map(|tp| {
+                    let mut mut_string = MutString::new(tp);
+                    replace_ss58_addresses(tp, &mut mut_string, &mut Default::default());
+                    mut_string.freeze()
+                })
+                .map(|tp| {
+                    let mut lexer = Lexer::new(&tp, "tp", Default::default());
+                    lexer
+                        .advance()
+                        .map_err(|err| Error::msg(format!("{:?}", err)))?;
+                    parse_type_params(&mut lexer)
+                })
+                .collect::<Result<_, _>>()?;
+        }
+        Ok(self)
+    }
+    /// Set the "args" based on the transmitted data from cmd
+    /// None => The value of None is ignored. Not assigned to a parameter
+    pub fn set_args_from_cmd(&mut self, args: Option<Vec<String>>) -> &mut Self {
+        if let Some(args) = args {
+            self.args = args
+                .iter()
+                .map(|arg| {
+                    let mut mut_string = MutString::new(arg);
+                    replace_ss58_addresses(arg, &mut mut_string, &mut Default::default());
+                    mut_string.freeze()
+                })
+                .collect();
+        }
+        self
+    }
+    /// Set the "signers" based on the transmitted data from cmd
+    /// None => The value of None is ignored. Not assigned to a parameter
+    pub fn set_signers_from_cmd(
+        &mut self,
+        signers: Option<Vec<String>>,
+    ) -> Result<&mut Self, Error> {
+        let mut signers = signers
+            .unwrap_or_default()
+            .iter()
+            .map(|signer| self.dove_ctx.dialect.parse_address(signer))
+            .collect::<Result<Vec<_>, Error>>()?;
+        if signers.is_empty() {
+            signers.push(self.dove_ctx.account_address()?);
+        }
+        self.signers = signers;
+        Ok(self)
+    }
+
+    // =============================================================================================
+    /// Create transaction
+    /// Return: Ok(name_transaction, Transaction)
+    pub fn to_transaction(&self) -> Result<(String, Transaction), Error> {
+        let (script, meta) = self.lookup_script()?;
+        let units = self.build_script(script)?;
+
+        let unit = units
+            .into_iter()
+            .find(|unit| {
+                let is_module = match &unit {
+                    CompiledUnit::Module { .. } => false,
+                    CompiledUnit::Script { .. } => true,
+                };
+                is_module && unit.name() == meta.name
+            })
+            .map(|unit| unit.serialize())
+            .map(|mut unit| {
+                self.dove_ctx
+                    .dialect
+                    .adapt_to_target(&mut unit)
+                    .map(|_| unit)
+            })
+            .ok_or_else(|| anyhow!("Script '{}' not found", meta.name))??;
+
+        if meta.type_parameters.len() != self.type_parameters.len() {
```

Comment:
nit: not necessary, but can be replaced with `anyhow::ensure!`

Reply:
Same on https://github.com/pontem-network/move-tools/pull/66/files#diff-1029745c2cf9c98df6d1b76587d311c216c5326caa39c860d4055d65f2b54c79R179

Reply:
I didn't understand

Reply:
[`ensure`](https://docs.rs/anyhow/1.0.41/anyhow/macro.ensure.html) is a neat macros which allows you to express all these checks in a more compact code:
Instead of 
```rust
if <some_condition> {
     return Err(anyhow!("My error message with {} params", params));
}
```
you can use
```rust
ensure!(<some_condition>, "My error message with {} params", params);
```

There are a lot of such checks in your code, so I think it can help to reduce code size slightly.